reactor-core/src/main/java/reactor/core/publisher/StrictSubscriber.java

```diff
@@ -72,9 +72,22 @@ public void onSubscribe(Subscription s) {
 			actual.onSubscribe(this);
 
 			if (Operators.setOnce(S, this, s)) {
-				long r = REQUESTED.getAndSet(this, 0L);
-				if (r != 0L) {
+				for (;;) {
```

Comment:
was this intended as a separate change?
if not, would benefit from the same level of commenting and from test coverage ;)

Reply:
Oh. That was for another PR